File: LibGemcadFileReader/Models/Geometry/Primitive/Triangle.cs
namespace LibGemcadFileReader.Models.Geometry.Primitive
{
    public class Triangle : Polygon
    {
        public Triangle()
            : base(3)
        {
        }

        public PolygonVertex P1
        {
            get
            {
                return Vertices[0];
            }
            set
            {
                Vertices[0].Assign(value);
            }
        }

        public PolygonVertex P2
        {
            get
            {
                return Vertices[1];
            }
            set
            {
                Vertices[1].Assign(value);
            }
        }

        public PolygonVertex P3
        {
            get
            {
                return Vertices[2];
            }
            set
            {
                Vertices[2].Assign(value);
            }
        }

        protected override bool PointCountImmutable => true;
    }
}

File: TestViewer/ClientApp/src/app/models/triangle.ts
import {Polygon} from "./polygon";
import {PolygonVertex} from "./polygon-vertex";

export class Triangle extends Polygon {
  public get p1(): PolygonVertex
  {
    return this.vertices[0];
  }

  public set p1(v: PolygonVertex)
  {
    this.vertices[0] = v;
  }

  public get p2(): PolygonVertex
  {
    return this.vertices[1];
  }

  public set p2(v: PolygonVertex)
  {
    this.vertices[1] = v;
  }

  public get p3(): PolygonVertex
  {
    return this.vertices[2];
  }

  public set p3(v: PolygonVertex)
  {
    this.vertices[2] = v;
  }
}
